### PR TITLE
Fix the error which gets thrown when the source files of a project contains quoted identifiers

### PR DESCRIPTION
**Pull request by miyurud**
## Purpose
This fixes the github [issue](https://github.com/ballerina-platform/ballerina-lang/issues/26109).

## Goals
Allow quoted identifiers also to be incorporated into the JSON files generated by the data mapper compiler plugin.

## Approach
Use URL encoding to convert the special characters into a string format that can be written into files.

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
JDK 1.8
 
## Learning
N/A

### CODE REVIEW

**Review by suganyasuven — APPROVED** [review]

File: src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java [hunk 1]
```diff
@@ -579,4 +589,19 @@ private String getCustomizedErrorMessage(Exception e) {
 
         return errorMessage;
     }
+
+    private static boolean hasAcceptedCharacters(String input) {
+        return input != null && input.matches("^[a-zA-Z0-9_:/.]*$");
```
Comment (rdhananjaya) [inline src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java:594]: Shall we use a compiled regex here. https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html

File: src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java [hunk 2]
```diff
@@ -140,18 +142,22 @@ public void process(PackageNode packageNode) {
                     DataMapperStructureVisitor visitor = new DataMapperStructureVisitor(structureMap);
                     visitor.visit(recordTypeDef);
                     String typeName = null;
+                    String serialized = null;
                     if (recordTypeDef.symbol instanceof  BRecordTypeSymbol) {
                         typeName = ((BRecordTypeSymbol) recordTypeDef.symbol).toString();
+                        String encodedTypeName = typeName;
+                        if (!hasAcceptedCharacters(encodedTypeName)) {
```
Comment (rdhananjaya) [inline src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java:149]: Do we need to encode this conditionally? IINM if the string doesn't contain any charactor that does not support in a URL then the original string will not change when we URLEncode it.

Comment (miyurud) [reply src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java:149]: I removed the conditional URL encoding logic and updated the code.

File: src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java (hunk 1 above)
Comment (miyurud) [reply src/main/java/org/ballerinax/datamapper/DataMapperPlugin.java:594]: With the latest improvements made to the code we will remove this method `hasAcceptedCharacters(String input)`

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**